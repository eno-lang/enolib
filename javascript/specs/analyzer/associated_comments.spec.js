const { analyze } = require('./analyze_helper.js');

// TODO: Re-check if snapshots still produce null

const input = `
> [comment]
> [comment]

> [comment]

> [comment_associated_with_list]
my_list:

> [comment_associated_with_list_item]
- My list item value
> [comment]

> [comment]
> [comment]

> [comment_associated_with_fieldset]
my_fieldset:

> [comment_associated_with_fieldset_entry_begin]
> [comment_associated_with_fieldset_entry_between]
> [comment_associated_with_fieldset_entry_end]
my_fieldset_entry = My fieldset entry value

> [comment_associated_with_section_begin]
> [comment_associated_with_section_end]
# my_section
`.trim();

describe('Associated comment analysis', () => {
  it('performs as specified', () => {
    expect(analyze(input)).toMatchSnapshot();
  });
});
