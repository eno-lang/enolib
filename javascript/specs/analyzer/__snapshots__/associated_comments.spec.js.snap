// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Associated comment analysis performs as specified 1`] = `
Object {
  "_document": Object {
    "depth": 0,
    "elements": Array [
      Object {
        "comments": Array [
          Object {
            "comment": "[comment_associated_with_list]",
            "line": 5,
            "ranges": Object {
              "comment": Array [
                40,
                70,
              ],
              "commentOperator": Array [
                38,
                39,
              ],
              "line": Array [
                38,
                70,
              ],
            },
            "type": Symbol(Comment),
          },
        ],
        "items": Array [
          Object {
            "comments": Array [
              Object {
                "comment": "[comment_associated_with_list_item]",
                "line": 8,
                "ranges": Object {
                  "comment": Array [
                    83,
                    118,
                  ],
                  "commentOperator": Array [
                    81,
                    82,
                  ],
                  "line": Array [
                    81,
                    118,
                  ],
                },
                "type": Symbol(Comment),
              },
            ],
            "continuations": Array [],
            "line": 9,
            "parent": [Circular],
            "ranges": Object {
              "itemOperator": Array [
                119,
                120,
              ],
              "line": Array [
                119,
                139,
              ],
              "value": Array [
                121,
                139,
              ],
            },
            "type": Symbol(Item),
            "value": "My list item value",
          },
        ],
        "key": "my_list",
        "line": 6,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            78,
            79,
          ],
          "key": Array [
            71,
            78,
          ],
          "line": Array [
            71,
            79,
          ],
        },
        "type": Symbol(List),
      },
      Object {
        "comments": Array [
          Object {
            "comment": "[comment_associated_with_fieldset]",
            "line": 15,
            "ranges": Object {
              "comment": Array [
                180,
                214,
              ],
              "commentOperator": Array [
                178,
                179,
              ],
              "line": Array [
                178,
                214,
              ],
            },
            "type": Symbol(Comment),
          },
        ],
        "entries": Array [
          Object {
            "comments": Array [
              Object {
                "comment": "[comment_associated_with_fieldset_entry_begin]",
                "line": 18,
                "ranges": Object {
                  "comment": Array [
                    231,
                    277,
                  ],
                  "commentOperator": Array [
                    229,
                    230,
                  ],
                  "line": Array [
                    229,
                    277,
                  ],
                },
                "type": Symbol(Comment),
              },
              Object {
                "comment": "[comment_associated_with_fieldset_entry_between]",
                "line": 19,
                "ranges": Object {
                  "comment": Array [
                    280,
                    328,
                  ],
                  "commentOperator": Array [
                    278,
                    279,
                  ],
                  "line": Array [
                    278,
                    328,
                  ],
                },
                "type": Symbol(Comment),
              },
              Object {
                "comment": "[comment_associated_with_fieldset_entry_end]",
                "line": 20,
                "ranges": Object {
                  "comment": Array [
                    331,
                    375,
                  ],
                  "commentOperator": Array [
                    329,
                    330,
                  ],
                  "line": Array [
                    329,
                    375,
                  ],
                },
                "type": Symbol(Comment),
              },
            ],
            "continuations": Array [],
            "key": "my_fieldset_entry",
            "line": 21,
            "parent": [Circular],
            "ranges": Object {
              "entryOperator": Array [
                394,
                395,
              ],
              "key": Array [
                376,
                393,
              ],
              "line": Array [
                376,
                419,
              ],
              "value": Array [
                396,
                419,
              ],
            },
            "type": Symbol(Attribute),
            "value": "My fieldset entry value",
          },
        ],
        "key": "my_fieldset",
        "line": 16,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            226,
            227,
          ],
          "key": Array [
            215,
            226,
          ],
          "line": Array [
            215,
            227,
          ],
        },
        "type": Symbol(Fieldset),
      },
      Object {
        "comments": Array [
          Object {
            "comment": "[comment_associated_with_section_begin]",
            "line": 23,
            "ranges": Object {
              "comment": Array [
                423,
                462,
              ],
              "commentOperator": Array [
                421,
                422,
              ],
              "line": Array [
                421,
                462,
              ],
            },
            "type": Symbol(Comment),
          },
          Object {
            "comment": "[comment_associated_with_section_end]",
            "line": 24,
            "ranges": Object {
              "comment": Array [
                465,
                502,
              ],
              "commentOperator": Array [
                463,
                464,
              ],
              "line": Array [
                463,
                502,
              ],
            },
            "type": Symbol(Comment),
          },
        ],
        "depth": 1,
        "elements": Array [],
        "key": "my_section",
        "line": 25,
        "parent": [Circular],
        "ranges": Object {
          "key": Array [
            505,
            515,
          ],
          "line": Array [
            503,
            515,
          ],
          "sectionOperator": Array [
            503,
            504,
          ],
        },
        "type": Symbol(Section),
      },
    ],
    "type": Symbol(Document),
  },
  "_lineCount": 26,
  "_meta": Array [
    Object {
      "comment": "[comment]",
      "line": 0,
      "ranges": Object {
        "comment": Array [
          2,
          11,
        ],
        "commentOperator": Array [
          0,
          1,
        ],
        "line": Array [
          0,
          11,
        ],
      },
      "type": Symbol(Comment),
    },
    Object {
      "comment": "[comment]",
      "line": 1,
      "ranges": Object {
        "comment": Array [
          14,
          23,
        ],
        "commentOperator": Array [
          12,
          13,
        ],
        "line": Array [
          12,
          23,
        ],
      },
      "type": Symbol(Comment),
    },
    Object {
      "comment": "[comment]",
      "line": 3,
      "ranges": Object {
        "comment": Array [
          27,
          36,
        ],
        "commentOperator": Array [
          25,
          26,
        ],
        "line": Array [
          25,
          36,
        ],
      },
      "type": Symbol(Comment),
    },
    Object {
      "comment": "[comment]",
      "line": 10,
      "ranges": Object {
        "comment": Array [
          142,
          151,
        ],
        "commentOperator": Array [
          140,
          141,
        ],
        "line": Array [
          140,
          151,
        ],
      },
      "type": Symbol(Comment),
    },
    Object {
      "comment": "[comment]",
      "line": 12,
      "ranges": Object {
        "comment": Array [
          155,
          164,
        ],
        "commentOperator": Array [
          153,
          154,
        ],
        "line": Array [
          153,
          164,
        ],
      },
      "type": Symbol(Comment),
    },
    Object {
      "comment": "[comment]",
      "line": 13,
      "ranges": Object {
        "comment": Array [
          167,
          176,
        ],
        "commentOperator": Array [
          165,
          166,
        ],
        "line": Array [
          165,
          176,
        ],
      },
      "type": Symbol(Comment),
    },
  ],
}
`;
